pkg/server/server.go

```diff
@@ -16,37 +17,44 @@ import (
 	pb "github.com/weaveworks/weave-gitops/pkg/api/applications"
 	"github.com/weaveworks/weave-gitops/pkg/kube"
 	"github.com/weaveworks/weave-gitops/pkg/middleware"
+	"github.com/weaveworks/weave-gitops/pkg/services/app"
+	"github.com/weaveworks/weave-gitops/pkg/utils"
 	"go.uber.org/zap"
+	"golang.org/x/oauth2"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type server struct {
+type key int
+
+const tokenKey key = iota
+
+type applicationServer struct {
 	pb.UnimplementedApplicationsServer
 
-	kube kube.Kube
-	log  logr.Logger
+	log logr.Logger
+	app *app.App
 }
 
-// An ApplicationsConfig allows for the customization of an ApplicationsServer.
+// An ServerConfig allows for the customization of an ApplicationsServer.
 // Use the DefaultConfig() to use the default dependencies.
-type ApplicationsConfig struct {
-	Logger     logr.Logger
-	KubeClient kube.Kube
+type ServerConfig struct {
```

Comment:
I think the WeGO enterprise team is relying on this. This might break their integration. What was the reason for this change?

Reply:
I thought it was done because it only handled the application proto. I wanted to make it cover both. If thats not the case i will certainly change it back.